examples/synthesis_json.py

```diff
@@ -0,0 +1,152 @@
+#        notice, this list of conditions and the following disclaimer.
+#     2. Redistributions in binary form must reproduce the above copyright
+#        notice, this list of conditions and the following disclaimer in the
+#        documentation and/or other materials provided with the distribution.
+#     3. Neither the name of the copyright holder nor the names of
+#        its contributors may be used to endorse or promote products
+#        derived from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY
+# DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+# ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+'''Extract the optimal distributions for the following features of the population of neurons:
+   soma: radius
+   basal dendrites: n_neurites
+   apical dendrites: n_neurites
+   axons: n_neurites
+   '''
+
+from neurom import ezy
+from neurom import stats
+import numpy as np
+import argparse
+import json
+from collections import OrderedDict
+import os
+
+
+def parse_args():
+    '''Parse command line arguments'''
+    parser = argparse.ArgumentParser(
+        description='Morphology fit distribution extractor',
+        epilog='Note: Outputs json of the optimal distribution \
+                and corresponding parameters.')
+
+    parser.add_argument('datapath',
+                        help='Path to morphology data file or directory')
+
+    return parser.parse_args()
+
+
+def extract_data(data_path, feature, params=None):
+    '''Loads a list of neurons, extracts feature
+       and transforms the fitted distribution in the correct format.
+       Returns the optimal distribution, corresponding parameters,
+       minimun and maximum values.
+    '''
+    population = ezy.load_neurons(data_path)
+
+    if params is None:
+        params = {}
+
+    feature_data = [getattr(n, 'get_' + feature)(*params) for n in population]
+
+    results = {}
+
+    try:
+        opt_fit = stats.optimal_distribution(feature_data)
+    except ValueError:
+        from itertools import chain
+        feature_data = list(chain(*feature_data))
+        opt_fit = stats.optimal_distribution(feature_data)
+
+    results["type"] = opt_fit[0]
+    results["params"] = opt_fit[1][0]
+    results["min"] = np.min(feature_data)
```

Comment:
I think we need a flag to say if we want to keep min and max. It doesn't always make sense to take these from the data (they could for example mark biological or physical boundaries). But if they get put in the file, they get used. Technically, I think min and max should be used as input to the fit, not be an output.